lib/settings.coffee

```diff
@@ -15,9 +15,34 @@ Settings =
       unless chrome.runtime.lastError
         @handleUpdateFromChromeStorage key, value for own key, value of items
 
+        @loadSettingsFromLocalStorage()
+
       chrome.storage.onChanged.addListener (changes, area) =>
         @propagateChangesFromChromeStorage changes if area == "sync"
 
+  loadSettingsFromLocalStorage: ->
+    chrome.storage.local.get null, (items) =>
+      unless chrome.runtime.lastError
+        overriddenBySync = []
+        for own key, value of items
+          if @shouldSyncKey key
+            if key of @cache
+              overriddenBySync.push key # This key has already been set from chrome.storage.sync.
+            else
+              @handleUpdateFromChromeStorage key, value
+
+        # All of the values in overriddenBySync have equivalents in chrome.storage.sync which take
+        # priority, so we remove them.
+        chrome.storage.local.remove overriddenBySync if Utils.isBackgroundPage()
+
+      chrome.storage.onChanged.addListener (changes, area) =>
+        return unless area == "local"
+        for key, change of changes
```

Comment:
I don't see how this is ever reached.  (There would have to be a `chrome.storage.local.set` somewhere, and I don't see one.  Or is this where you pick up the `localStorageClone` updates?

However, more importantly:

- I don't see `chrome.storage.local.remove()` anywhere.  And we *have to* remove these because, if the setting is reset to its default value, then an old (non-default) value from `chrome.storage.local` will take precedence, which is incorrect.

- The only mechanism for pushing options-page changes to content pages is `chrome.storage.sync`. When a setting which is not stored in `chrome.storage.sync` is changed back to its default value, there will be no `chrome.storage.sync` update, so such changes won't propagate to content pages.